Remotion/Web/UnitTests/Core/UI/Controls/ControlReplacing/StateReplacingStrategyTest.cs

```diff
@@ -59,28 +60,24 @@ public void LoadControlState ()
       _replacer.StateModificationStrategy = _stateModificationStrategy;
       _replacer.Controls.Add (testPageHolder.NamingContainer);
 
-      MemberCallerMock.Expect (mock => mock.SetChildControlState (Arg<ControlReplacer>.Is.Same (_replacer), Arg<Hashtable>.Is.NotNull));
-      MockRepository.ReplayAll ();
+      MemberCallerMock.Setup (mock => mock.SetChildControlState (_replacer, It.IsNotNull<Hashtable>())).Verifiable();
 
-      _stateModificationStrategy.LoadControlState (_replacer, MemberCallerMock);
-
-      MockRepository.VerifyAll ();
+      _stateModificationStrategy.LoadControlState (_replacer, MemberCallerMock.Object);
     }
 
     [Test]
     public void LoadViewState ()
     {
+      // TODO: Fix strrict mock setup, "IInternalControlMemberCaller.LoadViewStateRecursive(ControlReplacer, object) invocation failed with mock behavior Strict. All invocations on the mock must have a corresponding setup."
```

Comment:
TODO

Reply:
i think in some tests the setup is wrong.
e.g. in this test the mock expects two arguments, `_replacer` and any argument of type `Hashtable`
`MemberCallerMock.Setup (mock => mock.LoadViewStateRecursive (_replacer, It.IsNotNull<Hashtable>())).Verifiable();`

however when the mocked method is called, the arguments are not matching, because `_viewState` is of type `object` and not of type `Hashtable`
`memberCaller.LoadViewStateRecursive (replacer, _viewState);`

is it safe to just change `It.IsNotNull<Hashtable>` to `It.IsNotNull<object>`?

Reply:
Ok I just  found out why Rhino.Mocks behaves like this. If the type of the parameter is `object` Rhino.Mocks skips the type check.🥴🤯